### DISCUSSION TITLE
🎼 Master B major to control F major: simulation of, by, for Box-Flow

### DISCUSSION THREAD

**Discussion by hyunjimoon** [Data4DM/BayesSD#25] [perspectives; sense-making]
## Are we confusing MICRO and MACRO world?
I think we are applying probability sum law (mutual exclusive) and multiplication law (independent) in the same place but shouldn't..

"AR = Adoption from word of mouth + Adoption from advertisement"
<img width="549" alt="image" src="https://user-images.githubusercontent.com/30194633/193249030-80577e2d-dd29-4682-a46f-c538b998a81b.png">

Hazhir explained in infinitesimal time, one person cannot adopt both from advertisement and mouth. However, why does 'one person' view appears all of a sudden?

 And might this be relevant to problem "p4 which is a mixture is not the same as p3" Holes in Bayes related to quantum physics described below? 

<img width="465" alt="image" src="https://user-images.githubusercontent.com/30194633/192873268-e17148b0-5f89-45c2-a4ab-800f91226736.png">

## box-flow-box VS flow-box-flow?
Flow_major: faster = open system =  flow-box-flow = information, soul (move) = well-mixed = micro (= better with compartment model?)
Box_major: slower = closed system = box-flow-box = material, body (regulate) = less-mixed = macro  (= better with agent model?)

"Jules Verne: The body regulates the soul, and, like the balance-wheel, it is submitted to regular oscillations."

In the following model Repair_Inven, 
<img width="951" alt="image" src="https://user-images.githubusercontent.com/30194633/193251727-ecc8b61d-487e-4637-b5a6-400b51473ad8.png">

`Repair` is a Box_major and markov chain while `Inven` is Flow_major.  When the two are put in one model, `Repair` seems to be ever-lasting as it is compared against `Inven` which has shorter time-horizon (i.e. or may I say less sustainable? not that equilibrium is impossible in `Inven` system, but compared to `Repair` system it is. Hazhir mentioned finding equilibrium in Project model is useless because that would mean the project has ended and I think it might be relevant..)

However, I am no sure yet how to understand the two different formulation of SIR model (above Tom's, below: Hazhir's)
<img width="1076" alt="image" src="https://user-images.githubusercontent.com/30194633/193252014-ec09d7df-aac8-4822-a443-4d521260be3b.png">

More explanation on box-flow-box vs flow-box-flow and openness of system in https://github.com/Data4DM/BayesSD/blob/master/DailyDigest/4%20System%20Structures.md#3-openness-informationenergymaterial

**Comment by hyunjimoon** [comment]
## + is a symbol
In a higher probability space "+" is not the same any more. Fuzzy min and max.
- Transition between Discrete and Continuous time Markov chain is not straightforward. 
Compartment models is mixing MICRO_WORLD (rate addition (based on independence) with MACRO_WORLD (integration, ratio

- Deterministically non-existing but stochastically existing is my conclusion.. Much information is lost upon wlog z1 > z2 which can be overcome with distributional equivalence in stochastic design

<img width="617" alt="image" src="https://user-images.githubusercontent.com/30194633/193244984-c769de02-5844-4859-813d-54741cf5837a.png">


## Discrete and Continuous
<img width="812" alt="image" src="https://user-images.githubusercontent.com/30194633/193244555-0c78800b-c1af-4091-9ebd-cc4a999c48c9.png">
<img width="929" alt="image" src="https://user-images.githubusercontent.com/30194633/193246757-516c7f8c-c013-4340-a7dd-b77e5d10b5a4.png">


<img width="1254" alt="image" src="https://user-images.githubusercontent.com/30194633/193246676-db540be8-8f64-4972-a4fe-fe3ee83dc32d.png">
discrete and continuous
[code](https://github.com/hyunjimoon/defense-reliability/blob/17ad362d92e91f508ada9062b9f2a4ae5c2cd01e/twin/RepairInven.ipynb)

<img width="946" alt="image" src="https://user-images.githubusercontent.com/30194633/192872835-4d9cbfed-a70c-4ea0-8541-780919859c50.png">

**Reply by tomfid** [reply]
Not sure what the minimax thing is about, but the CES production function is a soft-min that approaches the min function in the limit. https://math.stackexchange.com/questions/604216/how-ces-function-becomes-minimum-function

**Comment by hyunjimoon** [comment]
Functional formulation, coflow (how to overcome the independent assumption or full mixing), allocation may be relevant of rainbow between C and A (will call California rainbow :))
Hazhir has a paper 
<img width="399" alt="image" src="https://user-images.githubusercontent.com/30194633/193306300-c3707e5d-77e8-400a-ba18-e872fddea01f.png">

[System Dynamics Review - 2010 - Rahmandad - Modeling the rework cycle  capturing multiple defects per task.pdf](https://github.com/Data4DM/BayesSD/files/9685881/System.Dynamics.Review.-.2010.-.Rahmandad.-.Modeling.the.rework.cycle.capturing.multiple.defects.per.task.pdf)

**Comment by tomfid** [comment]
Lots of questions here.

I think the "one person" perspective arises because a person is the natural atomic agent in this system. The additivity is a consequence of the standard DES/queuing result that any sufficiently short interval has vanishing probability of more than one Poisson event.

However, that's contingent on the simple model of causality depicted in the Bass model: a contact with a user or ad instantly induces conversions (with some probability). If you built an agent version of the model with individuals motivated in that way, and well-mixed contacts, you should get exactly the same result.

I'm sure it's possible to construct a different model of the individual, where for example P(conversion) increases in a nonlinear way with accumulated exposure. That would potentially be much harder to aggregate. OTOH in a lot of models, the difference might not be important.

**Comment by hyunjimoon** [comment]
@tomfid from Nathaniel's ["characterizing heterogeneity in agent based modeling" ](https://youtu.be/KBaECxmZ5LE), the following slides caught my eyes

**Reply by hyunjimoon** [reply]
![image](https://user-images.githubusercontent.com/30194633/194549815-fc1e12fe-5689-4b43-bccd-6779eb90678a.jpeg)
![image](https://user-images.githubusercontent.com/30194633/194549840-2fd4ac26-6e86-44e0-8e53-6acfd90a721b.jpeg)

**Comment by hyunjimoon** [comment]
Sustainability targets F major, but B-major should be mastered in advance:
 i) capability
 ii) perceived drum for sync signal of system components 

![image](https://user-images.githubusercontent.com/30194633/202914305-d0d17e62-c91f-4b72-a79f-b11776010517.png)

**Comment by hyunjimoon** [comment]
From 879 seminar models [here](https://publish.obsidian.md/tolzul/%F0%9F%91%81%F0%9F%A7%A0%F0%9F%8C%B2%F0%9F%8F%AD+SimBasedCalib/deConv(SBC)/%F0%9F%8F%AD+Generator_1apx/categ(Generator)/eg(Generator)) I sorted by the number of box + flow.

| model no. |                 | sum | box                                                                                                                                                                                                                                      | flow                                                                                                                                                      |
| --------- | --------------- | --- | ---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- | --------------------------------------------------------------------------------------------------------------------------------------------------------- |
| 1         | SEIRb           | 11  | 6 suceptible, exposed, infected, recovered, dead, death_perceived                                                                                                                                                                        | 5 exposing, onsetting, recovering, dying, death_perceiving                                                                                                |
| 2         | project         | 12  | 6 project_resource, task_to_be(=backlog)-{done, reworked, quality_assuranced} , task_released, project_lasted                                                                                                                            | 6 resource_adjusting, completing, rework-{discovering,doing}, work_approving, resource_adjusting, dealine_delaying                                        |
| 3         | market growth   | 12  | 6 sales_force, recent_revenue, backlog, delivery_delay_market_preceived, delivery_delay_company_preceived, capacity                                                                                                                      | 6 salesforce_net_hiring, revenue_net_changing, backlog-{ordering,shiping}, capacity_net_acquisiting, delivery_delay_perceiving                            |
| 4         | long wave       | 12  | 5 order_expected, capital-{supply_line, stock}, goods_supply_line, backlog                                                                                                                                                               | 7 capital-{ordering,acquisiting,discarding}, capital_backlog-{ordering,producing}, goods_capital-{ordering,acquisiting}                                   |
| 5         | mat_info_price  | 17  | 8 production_cost_expected, industry_demanded, {long, short}-run_price_expected, variable_cost_expected, capacity_utilization, wip, inv                                                                                                  | 9 product-{starting, ing, shiping}, charm-{{capacity_util, industry_demand}-nonprice, {short, long}-price}-adjusting, cost-{expected, variable}-adjusting |
| 6         | service quality | 27  | 13 labor_desired, labor_productivity_perceived, vacancies, rookies, personnel_experienced, backlog_service, desired_to, fatigue-{A, E}, cust_service_expected, employee_quality_expected, mgmt_quality_perceived, cust_quality_perceived | 7 + 7                                                                                                                                                     |
